src/api.py

```diff
@@ -293,16 +364,22 @@ def run(
         """Run the text generator against all the text, combined"""
 
         self.config.extend_with_dict(request.data.options, overwrite=True)
-
+        requested_capabilities = RequestedCapabilities(SUPPORT_MAP)
+        capability_response = requested_capabilities.extract_from_blocks(request.data.blocks)
+        if not capability_response:
+            requested_capabilities = None
         messages = self.prepare_messages(request.data.blocks)
         if self.config.moderate_output and self._flagged(messages):
             raise SteamshipError(
                 "Sorry, this content is flagged as inappropriate by OpenAI."
             )
         user_id = self.context.user_id if self.context is not None else "testing"
         generated_blocks, usage_reports = self.generate_with_retry(
-            messages=messages, user=user_id, options=request.data.options
+            messages=messages, user=user_id, options=request.data.options, requested_capabilities=requested_capabilities
         )
+        if capability_response:
+            # TODO append to file?
```

Comment:
The plugin itself doesn't have to write anything to Files.  Where the blocks are saved is mediated by the engine.